content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,324 @@
+
+### Reducing performance impact
+
+If your application sends a large number of spans to a collector, this can have a significant performance impact. You can use the following approaches to reduce this impact:
+
+- [Use the BatchSpanProcessor](#send-traces-in-batches-using-the-batchspanprocessor)
+- [Send fewer spans to the collector](#send-fewer-spans-to-the-collector-via-sampling)
+
+#### Send Traces in Batches using the `BatchSpanProcessor`
+
+In a production environment, you can use the OpenTelemetry `BatchSpanProcessor` to send the spans to a collector in batches rather than one at a time. However, during development and testing, you might not want to send spans in batches. In this situation, you might prefer to use the `SimpleSpanProcessor`.
+
+You can configure your tracing configuration to use the appropriate span processor, depending on the environment, as follows:
+
+```ts
+import {
+  SimpleSpanProcessor,
+  BatchSpanProcessor,
+} from '@opentelemetry/sdk-trace-base'
+
+if (process.env.NODE_ENV === 'production') {
+  provider.addSpanProcessor(new BatchSpanProcessor(otlpTraceExporter))
+} else {
+  provider.addSpanProcessor(new SimpleSpanProcessor(otlpTraceExporter))
+}
+```
+
+#### Send fewer spans to the collector via sampling
+
+Another way to reduce the performance impact is to [use probability sampling](https://opentelemetry.io/docs/reference/specification/trace/tracestate-probability-sampling/) to send fewer spans to the collector. This reduces the collection cost of tracing but still gives a good representation of what is happening in your application.
+
+An example implementation looks like this:
+
+```ts highlight=4;
+import { TraceIdRatioBasedSampler } from '@opentelemetry/core'
+
+const provider = new BasicTracerProvider({
+  sampler: new TraceIdRatioBasedSampler(0.1),
+  resource: new Resource({
+    // we can define some metadata about the trace resource
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+```
+
+## Troubleshooting
+
+### My traces aren't showing up
+
+The order in which you setup tracing matters. You'll want to make sure you've registered tracing and setup instrumentation before importing instrumented dependencies. For example:
+
+```ts
+import { registerTracing } from './tracing'
+
+registerTracing({
+  name: 'tracing-example',
+  version: '0.0.1',
+})
+
+// Should come after you've setup tracing.
+import { PrismaClient } from '@prisma/client'
+import async from 'express-async-handler'
+import express from 'express'
+```
+
+### Child traces start before parent traces
+
+We're still investigating [this issue](https://github.com/prisma/prisma/issues/14612).
+
+### Some traces show 0μs
```

Comment:
(Or something similar... some context for '0μs'

```suggestion
### Some traces show a timing measurement of 0μs
```

Reply:
This is covered!